editor/src/clj/editor/engine.clj

```diff
@@ -208,24 +208,23 @@
       (fs/set-executable! engine-file)
       engine-file)))
 
-(defn- zip-entries! [zipfile]
+(defn- zip-entries! [^ZipFile zipfile]
   (enumeration-seq (.entries zipfile)))
 
 (defn- unpack-build-zip!
   [^File engine-archive dir]
   (with-open [zip-file (ZipFile. engine-archive)]
-    (doseq [entry (zip-entries! zip-file)]
+    (doseq [^ZipEntry entry (zip-entries! zip-file)]
       (let [savePath (str dir File/separatorChar (.getName entry))
             saveFile (File. savePath)]
         (if (.isDirectory entry)
           (if-not (.exists saveFile)
             (.mkdirs saveFile))
           (let [parentDir (File. (.substring savePath 0 (.lastIndexOf savePath (int File/separatorChar))))
                 stream (.getInputStream zip-file entry)]
-            (if-not (.exists parentDir) (.mkdirs parentDir))
-            (clojure.java.io/copy stream saveFile)))
-      ))))
-
+            (when-not (.exists parentDir)
+              (.mkdirs parentDir))
+            (io/copy stream saveFile)))))))
```

Comment:
Fixed syntax. Also, we try to use `when` instead of `if` for side-effects. `ìf` should be seen as the ternary operator (I.e. `true ? 1 : 0`), because it always returns a value.